### DISCUSSION TITLE
Need more documentation on how are preimages generated and if there are any limitations with static channel backups and HOLD invoices

### DISCUSSION THREAD

**Discussion by AndySchroder** [lightningnetwork/lnd#6696] [Q&A] [answered]
How are preimages generated? Is the randomness sourced from the operating system every time a new preimages is needed? Or, are they deterministiclly generated from the random seed words you need to backup when first creating an lnd instance?

It seems with the static channel backups, there must be some deterministic nature of the preimage generation.

However, HOLD invoices require the user to manually generate their own preimage and hash.

Trying to understand why there is not an option with HOLD invoices to let LND generate the preimage for me so that I don't need to make sure I am doing it in a secure way. Also, wondering if there are any limitations with static channel backups if using HOLD invoices.

Would be nice if the answers to these questions were documented somewhere.

**Comment by alexbosworth** [comment, marked as answer]
There is a discussions feature in github for topics q&a you could check out

If you just generate a secure random 32 byte value for your hold invoice preimage that is secure and no it has no impact on static channel backups
